frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -188,6 +219,94 @@ function MainMenu(props: Props): ReactElement {
   const { colors }: Theme = useTheme()
   const isServerDisconnected = !props.isServerConnected
 
+  const onClickDeployApp = useCallback((): void => {
+    const {
+      showDeployError,
+      closeDialog,
+      isDeployErrorModalOpen,
+      gitInfo,
+    } = props
+
+    if (!gitInfo) {
+      const dialog = NoRepositoryDetected()
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    const {
+      repository,
+      branch,
+      module,
+      untrackedFiles,
+      uncommittedFiles,
+      state: gitState,
+    } = gitInfo
+
+    if (
+      (!repository || !branch || !module) &&
```

Comment:
Nit: Can we move `!repository || !branch || !module` part into a variable? Multi-line `if`s are hard to read.